actor/src/main/scala/org/apache/pekko/io/dns/internal/DnsClient.scala

```diff
@@ -140,9 +157,25 @@ import pekko.pattern.{ BackoffOpts, BackoffSupervisor }
             log.debug("Client for id {} not found. Discarding unsuccessful response.", msg.id)
         }
       } else {
-        val (recs, additionalRecs) =
-          if (msg.flags.responseCode == ResponseCode.SUCCESS) (msg.answerRecs, msg.additionalRecs) else (Nil, Nil)
-        self ! Answer(msg.id, recs, additionalRecs)
+        inflightRequests.get(msg.id) match {
+          case Some((_, orig)) if !isSameQuestion(msg.questions, orig.questions) =>
```

Comment:
There was a double bug here.  First the if expression was missing the `!`, then `isSameQuestion` was matching on a `Seq` with `List` extractors. The double negative caused the test to pass.